### DISCUSSION TITLE
Overlay Header on the same level as Modal

### DISCUSSION THREAD

**Discussion by k1ng-slayer** [callstack/react-native-paper#4342] [Q&A]
i want to elevate my Header? How can i do that?

**Comment by tamish-max** [comment]
If the header is rendered normally in the screen tree, it will not sit on the same layer as a Paper `Modal`.

`Modal` in React Native Paper is rendered through a `Portal`, so it is intentionally above the regular React tree. That means `zIndex` on your normal header usually will not help.

Two workable options:

1. render the header inside the same `Portal` as the modal content, or
2. move both the header and modal into a shared overlay layer that you control.

So if the goal is “header above modal” or “header at the same overlay level”, the fix is architectural: the header also needs to live in a portal/overlay layer.
